src/testing/naca0012_unsteady_check_quick.cpp

```diff
@@ -0,0 +1,75 @@
+#include "naca0012_unsteady_check_quick.h"
+#include "flow_solver/flow_solver_factory.h"
+#include "flow_solver/flow_solver_cases/naca0012.h"
+
+namespace PHiLiP {
+namespace Tests {
+
+template <int dim, int nstate>
+NACA0012UnsteadyCheckQuick<dim, nstate>::NACA0012UnsteadyCheckQuick(
+    const PHiLiP::Parameters::AllParameters *const parameters_input,
+    const dealii::ParameterHandler &parameter_handler_input)
+        : TestsBase::TestsBase(parameters_input)
+        , parameter_handler(parameter_handler_input)
+{}
+
+template <int dim, int nstate>
+int NACA0012UnsteadyCheckQuick<dim, nstate>::run_test() const
+{
+    double lift_expected;
+    double drag_expected;
+    const double acceptable_tolerance = 0.00001;
+
+    // These values are hard-coded from the results of running the test case.
+    // It is expected that changes such as boundary implementation,
+    // flux choice etc could change these values, so they may need to be updated
+    // in the future.
+    if (this->all_parameters->use_weak_form) {
+        // Weak, Roe flux
+        lift_expected = 0.198245;
+        drag_expected = 0.220394;
+    }else if (this->all_parameters->use_split_form) {
+        // Strong, Ra two-point flux
+        lift_expected = 0.24326;
+        drag_expected = 0.268897;
+    } else {
+        // Strong, Roe flux
+        lift_expected = 0.198245;
+        drag_expected = 0.220394;
+    }
```

Comment:
If we're going to use magic numbers that depend on the number of steps, step size, final time, numerical flux, etc. they should also be hardcoded here. That way, if someone changes one thing, such as the number of steps, they see the surrounding magic numbers that would also be affected. Therefore, the above should specify any non-default parameters. The parameter file should contain variations for which this test passes, (e.g. weak form, strong form, strong Ra two-point flux).

I also see that both strong and weak use the same expected changes. If they are expected to be the same, let's use 1 set of magic numbers. Also, what makes us believe that the `Strong, Ra two-point flux` is correct? If `Strong, Ra two-point flux` starts failing because of a different lift without any obvious change, how do we know that the change was incorrect or that the old version was incorrect?

Alternatively to all the above comments, I think that a more robust test here would be to run both strong and weak DG, and then compare the lift and drag, or any other expected observable measure. That way, the test doesn't depend on magic values and works for any parameter input. The intent of the test is therefore "whatever weak DG does, strong DG does similarly as well", if nothing else breaks, one of the two is wrong.

Reply:
Responding to each paragraph:
1. I structured the test similarly to the TGV energy tests, which specify an expected energy in the prm file ([e.g](https://github.com/dougshidong/PHiLiP/blob/a29ecce53f33a3f0bafc14ef8db767fe967e29b8/tests/integration_tests_control_files/taylor_green_vortex_integration/viscous_taylor_green_vortex_energy_check_strong_quick.prm#L66) ). I think the approach in those tests is a bit better because the expected energy is in the same .prm file as all other parameters. However, since I don't see this test being re-used extensively and I wanted to avoid adding more test-specific parameters. I recognize that this isn't ideal in terms of coding practice, and I can change it if needed.
2. At this time step size, strong and weak yield the same lift and drag, which I believe we should expect for a subsonic flow per the discussion on discord. I can change it to be a single condition, checking `if ( (strong || weak) && not split) `.  I didn't do this initially as when the time step was bigger, they had different values. Re: "What makes us believe that `Strong, Ra two-point flux` is correct?", I anticipate that the current values may change with Julien's changes to adiabatic wall boundaries -- see issue #240. The test will fail if there are changes which impact lift or drag. We won't know which version is correct without other tests, but this test will alert us to any major changes; if there is reason to believe a new version is correct, the hard-coded "magic numbers" will need to be changed. The console output should make it easy to recognize why a test is failing.
3. I agree that this would be a more robust way to compare weak and strong DG if the user is expected to be changing the parameters file. However, I don't think we can expect that split form strong DG will yield the same lift and drag as non-split DG, as the flux is different. Therefore, we would still need this sort of "magic number" test for split form strong, as there is no weak equivalent. 

I can come back to this PR either tomorrow or Friday. Of the following options, which would you prefer?
1. Keep the test as-is, with hard-coded lift and drag values. Remove the duplicate lift and drag such that there are only two sets.
2. Move the expected lift and drag into parameters, consistent with the TGV energy check tests. Having them in the same place as the test parameters will reduce the likelihood that a user changes a parameter and gets an unexpected testfail. However, this will mean more test-specific parameters.
3. Make an additional test which compares the lift and drag resulting from strong and weak DG with otherwise identical parameters. There will still be a hard-coded strong-split test.

Reply:
Update: I just pushed a new version of the test. It compares strong and weak with Roe flux, using a single test and no hard-coded numbers. There is no check for strong-split at this time, but I have a framework set up for it in the test.